extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -144,13 +202,16 @@ function BaseCarouselWithRef(
       )}
       <Scroller
         advanceCount={advanceCount}
+        autoAdvanceCount={autoAdvanceCount}
         loop={loop}
         mixedLength={mixedLength}
         restingIndex={currentSlide}
         setRestingIndex={setRestingIndex}
         snap={snap}
         ref={scrollRef}
-        onTouchStart={() => setHadTouch(true)}
+        onTouchStart={() => (interaction.current = Interaction.TOUCH)}
+        onMouseEnter={() => (interaction.current = Interaction.MOUSE)}
+        onFocus={() => (interaction.current = Interaction.FOCUS)}
```

Comment:
Does `onFocus` work here? Not sure how React/Preact treat this, but classically in DOM "focus" is a [non-bubbling event](https://developer.mozilla.org/en-US/docs/Web/API/Element/focus_event). If this were DOM, I'd be using `focusin` and `focusout` events that do bubble. Could you please confirm you see "focus" arriving?

Reply:
[onFocus](https://reactjs.org/docs/events.html#focus-events) is the recommended prop for listening. However this helped me realize this and the others are attached to the wrong element. :) I moved them to the outer most `ContainWrapper` and also opted to call any user provided versions so they do not get overwritten.